http-client/src/test/groovy/io/micronaut/http/client/config/DefaultHttpClientConfigurationSpec.groovy

```diff
@@ -78,6 +78,25 @@ class DefaultHttpClientConfigurationSpec extends Specification {
         'enabled'         | 'enabled'        | 'false' | false
     }
 
+    void "test WebSocket compression config"() {
+        given:
+        def ctx = ApplicationContext.run(
+                ("micronaut.http.client.ws.compression.$key".toString()): value
+        )
+        HttpClientConfiguration config = ctx.getBean(HttpClientConfiguration)
+        HttpClientConfiguration.WebSocketCompressionConfiguration compressionConfig = config.getWebSocketCompressionConfiguration()
+
+        expect:
+        compressionConfig[property] == expected
+
+        cleanup:
+        ctx.close()
+
+        where:
+        key               | property         | value   | expected
+        'enabled'         | 'enabled'        | 'false' | false
+    }
+
```

Comment:
Shouldn't we test the enabled 'true' setting?

And do we need a test for the default?

Reply:
Maybe? :) TBH, I was just following the existing patterns in this spec. IIRC every test in here is testing the config properties being set to something other than the default.

IMO, the more interesting (and effective) tests are in `BinaryWebSocketSpec` - we do test the default/true case there.

Reply:
Hehehe, yeah I think this spec isn't the best... Using a `where:` provider for a single case just adds an un-required level of indirection and confusion 😢 

Pushed https://github.com/micronaut-projects/micronaut-core/pull/10357/commits/492b6955b2a32fa762664d750ee20de20aaea8c7 to modernize and simplify things 😎